### ISSUE TITLE
[2.3.0-M03] Neo.ClientError.Statement.EntityNotFound when attempting to delete a path

### ISSUE THREAD

**Issue by jotomo** [neo4j/neo4j#5598]
In the hopes that the ability to delete paths is here to stay :-)
The error in the subject is from the browser, which is reproduced below with the shell.

```
$ cat del-path.cql 
match (n) return count(*);
create (:Test)-[:R]->(:Test);
match p=(n:Test)-[*]-() delete p;
match p=(n:Test)-[*]-() detach delete p; // isn't executed due to previous error, yields same result

$ neo4j-shell -path test.db -v -file del-path.cql 

+----------+
| count(*) |
+----------+
| 0        |
+----------+
1 row
984 ms
+-------------------+
| No data returned. |
+-------------------+
Nodes created: 2
Relationships created: 1
Labels added: 2
102 ms
ERROR (-v for expanded information):
    EntityNotFoundException: Unable to load NODE with id 1.
org.neo4j.kernel.api.exceptions.EntityNotFoundException: Unable to load NODE with id 1.
    at org.neo4j.kernel.impl.api.StateHandlingStatementOperations.nodeCursorById(StateHandlingStatementOperations.java:138)
    at org.neo4j.kernel.impl.api.ConstraintEnforcingEntityOperations.nodeCursorById(ConstraintEnforcingEntityOperations.java:394)
    at org.neo4j.kernel.impl.api.OperationsFacade.nodeGetRelationships(OperationsFacade.java:328)
    at org.neo4j.cypher.internal.spi.v2_3.TransactionBoundQueryContext.getRelationshipsForIds(TransactionBoundQueryContext.scala:135)
    at org.neo4j.cypher.internal.compiler.v2_3.spi.DelegatingQueryContext.getRelationshipsForIds(DelegatingQueryContext.scala:59)
    at org.neo4j.cypher.internal.compiler.v2_3.spi.DelegatingQueryContext.getRelationshipsForIds(DelegatingQueryContext.scala:59)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.VarLengthStep$$anonfun$1.apply(VarLengthStep.scala:87)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.VarLengthStep$$anonfun$1.apply(VarLengthStep.scala:87)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.DynamicIterable$$anon$1.iterator(DynamicIterable.scala:29)
    at scala.collection.IterableLike$class.isEmpty(IterableLike.scala:81)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.DynamicIterable$$anon$1.isEmpty(DynamicIterable.scala:28)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.VarLengthStep.expand(VarLengthStep.scala:90)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.TraversalPathExpander.expand(TraversalPathExpander.scala:37)
    at org.neo4j.kernel.impl.traversal.TraversalBranchWithState.expandRelationshipsWithoutChecks(TraversalBranchWithState.java:70)
    at org.neo4j.kernel.impl.traversal.TraversalBranchImpl.expandRelationships(TraversalBranchImpl.java:104)
    at org.neo4j.kernel.impl.traversal.StartNodeTraversalBranch.next(StartNodeTraversalBranch.java:47)
    at org.neo4j.kernel.impl.traversal.AsOneStartBranch.next(AsOneStartBranch.java:109)
    at org.neo4j.graphdb.traversal.PreorderDepthFirstSelector.next(PreorderDepthFirstSelector.java:49)
    at org.neo4j.kernel.impl.traversal.MonoDirectionalTraverserIterator.fetchNextOrNull(MonoDirectionalTraverserIterator.java:65)
    at org.neo4j.kernel.impl.traversal.MonoDirectionalTraverserIterator.fetchNextOrNull(MonoDirectionalTraverserIterator.java:36)
    at org.neo4j.helpers.collection.PrefetchingIterator.peek(PrefetchingIterator.java:60)
    at org.neo4j.helpers.collection.PrefetchingIterator.hasNext(PrefetchingIterator.java:46)
    at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:42)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:369)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.EmptyResultPipe.internalCreateResults(EmptyResultPipe.scala:30)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:125)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.createResults(DefaultExecutionResultBuilderFactory.scala:93)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.build(DefaultExecutionResultBuilderFactory.scala:63)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:230)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:214)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anon$3.run(ExecutionPlanBuilder.scala:172)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192)
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.runSafely(CompatibilityFor2_3.scala:114)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper.run(CompatibilityFor2_3.scala:191)
    at org.neo4j.cypher.internal.PreparedPlanExecution.execute(PreparedPlanExecution.scala:27)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:118)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:112)
    at org.neo4j.cypher.javacompat.internal.ServerExecutionEngine.executeQuery(ServerExecutionEngine.java:62)
    at org.neo4j.shell.kernel.apps.cypher.Start.getResult(Start.java:87)
    at org.neo4j.shell.kernel.apps.cypher.Start.exec(Start.java:68)
    at org.neo4j.shell.kernel.apps.TransactionProvidingApp.execute(TransactionProvidingApp.java:253)
    at org.neo4j.shell.impl.AbstractAppServer.interpretLine(AbstractAppServer.java:126)
    at org.neo4j.shell.kernel.GraphDatabaseShellServer.interpretLine(GraphDatabaseShellServer.java:96)
    at org.neo4j.shell.impl.AbstractClient.evaluate(AbstractClient.java:149)
    at org.neo4j.shell.impl.AbstractClient.evaluate(AbstractClient.java:133)
    at org.neo4j.shell.StartClient.executeCommandStream(StartClient.java:394)
    at org.neo4j.shell.StartClient.grabPromptOrJustExecuteCommand(StartClient.java:373)
    at org.neo4j.shell.StartClient.tryStartLocalServerAndClient(StartClient.java:274)
    at org.neo4j.shell.StartClient.startLocal(StartClient.java:247)
    at org.neo4j.shell.StartClient.start(StartClient.java:180)
    at org.neo4j.shell.StartClient.main(StartClient.java:135)

```

**Comment by jotomo** [comment]
2.3.1: still present
3.0.0-M01: fixed

**Comment by Mats-SX** [comment]
@jotomo Thanks a lot for reporting this. Yes, we do intend to keep the ability to delete paths around in Cypher. I've implemented a fix for it, which will be included in 2.3.3.
